src/cmd/ksh93/sh/path.c

```diff
@@ -1613,8 +1613,13 @@ void path_alias(Namval_t *np, Pathcomp_t *pp) {
         Shell_t *shp = sh_ptr(np);
         struct stat statb;
         char *sp;
+        Pathcomp_t *old = 0;
         nv_offattr(np, NV_NOPRINT);
         nv_stack(np, &talias_init);
+        old = (Pathcomp_t*)np->nvalue.cp;
```

Comment:
Since `np->nvalue.cp` is a `const char *` why are you declaring `old` to be a `Pathcomp_t *` thus necessitating this cast?

Reply:
We need to read `old->refcount` before free'ing the pointer.

Reply:
Done!